overlord/snapstate/snapstate_update_test.go

```diff
@@ -8761,6 +8770,241 @@ func failAfterLinkSnap(ol *overlord.Overlord, chg *state.Change) error {
+	restore := release.MockOnClassic(false)
+	defer restore()
+	restore = snapstate.MockRevisionDate(nil)
+	defer restore()
+
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	// make it easier for us to mock the whole gadget update so we don't
+	// have to jump through to many hoops. Streamline it a bit as well by
+	// making sure that one of the tasks requires a reboot by the link-snap
+	s.o.TaskRunner().AddHandler("update-gadget-assets",
+		func(task *state.Task, tomb *tomb.Tomb) error {
+			chg := task.Change()
+			chg.Set("gadget-restart-required", true)
+			return nil
+		},
+		func(task *state.Task, tomb *tomb.Tomb) error { return nil })
+
+	s.o.TaskRunner().AddHandler("update-gadget-cmdline",
+		func(task *state.Task, tomb *tomb.Tomb) error { return nil },
+		func(task *state.Task, tomb *tomb.Tomb) error { return nil })
+
+	var restartRequested []restart.RestartType
+	_, err := restart.Manager(s.state, "boot-id-0", snapstatetest.MockRestartHandler(func(t restart.RestartType) {
+		restartRequested = append(restartRequested, t)
+	}))
+	c.Assert(err, IsNil)
+
+	restore = snapstatetest.MockDeviceModel(ModelWithBase("core18"))
+	defer restore()
+
+	types := map[string]string{
+		"snapd":          "snapd",
+		"core18":         "base",
+		"gadget":         "gadget",
+		"kernel":         "kernel",
+		"some-base":      "base",
+		"some-base-snap": "app",
+	}
+	snapIds := map[string]string{
+		"snapd":          "snapd-snap-id",
+		"core18":         "core18-snap-id",
+		"gadget":         "gadget-core18-id",
+		"kernel":         "kernel-id",
+		"some-base":      "some-base-id",
+		"some-base-snap": "some-base-snap-id",
+	}
+
+	for _, sn := range order {
+		si := &snap.SideInfo{RealName: sn, Revision: snap.R(1), SnapID: snapIds[sn]}
+		snapYaml := fmt.Sprintf("name: %s\nversion: 1.2.3\ntype: %s", sn, types[sn])
+		if sn == "some-base-snap" {
+			// add a base for the app if the base is in the update list
+			snapYaml += "\nbase: some-base"
+		}
+
+		snaptest.MockSnap(c, snapYaml, si)
+		snapstate.Set(s.state, sn, &snapstate.SnapState{
+			Active:          true,
+			Sequence:        []*snap.SideInfo{si},
+			Current:         si.Revision,
+			TrackingChannel: "latest/stable",
+			SnapType:        types[sn],
+		})
+	}
+
+	chg := s.state.NewChange("refresh", fmt.Sprintf("refresh %v", order))
+	affected, tss, err := snapstate.UpdateMany(context.Background(), s.state,
+		order, nil, s.user.ID, &snapstate.Flags{NoReRefresh: true})
+	c.Assert(err, IsNil)
+	sort.Strings(affected)
+	c.Assert(affected, testutil.DeepUnsortedMatches, order)
+
+	findTs := func(sn string) *state.TaskSet {
+		for _, ts := range tss {
+			for _, tsk := range ts.Tasks() {
+				if tsk.Kind() == "prerequisites" {
+					snapsup, err := snapstate.TaskSnapSetup(tsk)
+					c.Assert(err, IsNil)
+					if snapsup.InstanceName() == sn {
+						return ts
+					}
+					break
+				}
+			}
+		}
+		return nil
+	}
+
+	// Map all relevant task-sets.
+	tsByOrder := make(map[string]*state.TaskSet)
```

Comment:
I don't understand the name of this variable, as the index is a name, not a position

Reply:
good point, renaming to `tsByName` as that is what it is